backend/tests/conftest.py

```diff
@@ -1,16 +1,45 @@
+import json
+import os
+from app.models.submission import Submission
+from app.models.user import User
 import pytest
 from sqlalchemy import create_engine
 from sqlalchemy.orm import sessionmaker
-from fastapi.testclient import TestClient
-import os
-
+from sqlalchemy.orm import Session
+from datetime import datetime
 from app.core.database import Base
+from pyrootutils import find_root
 from app.main import app
+from fastapi.testclient import TestClient
 from app.core.session import get_db
 
 
-@pytest.fixture(scope="function")
-def test_db():
+@pytest.fixture(scope="session")
+def test_users() -> list[User]:
+    data_path = find_root() / "tests" / "test_data" / "users.json"
+    with open(data_path, "r") as f:
+        users = json.load(f)
+
+    users = [User(**user) for user in users]
+    return users
+
+
+@pytest.fixture(scope="session")
+def test_submissions() -> list[Submission]:
+    data_path = find_root() / "tests" / "test_data" / "submissions.json"
+    with open(data_path, "r") as f:
+        submissions = json.load(f)
+
+    for submission in submissions:
+        submission["created_at"] = datetime.strptime(submission["created_at"], "%Y-%m-%d %H:%M:%S")
+        submission["updated_at"] = datetime.strptime(submission["updated_at"], "%Y-%m-%d %H:%M:%S")
+
+    submissions = [Submission(**submission) for submission in submissions]
+    return submissions
```

Comment:
`test_submission` will only work if `test_users` was launched before that, right? Because `test_submission` wants several users in the users table in the database.

I think that's a wrong pattern. if we modify one test, it should not affect other tests. And if we do them independent, we can keep `pytest.fixture(scope="function")`

Reply:
similarly, `test_leaderboard_db` requires previous tests. Maybe we can do a longer test like

```python
def test_submission_workflow():
    user_1 = User()
    user_2 = User()
    submission = Submission(user_1)
    submission = Submission(user_2)
    leaderboard = Submission.get_leaderboard()
    assert leaderboard
```

Reply:
same for API testing, I think it's fine to have a full flow, all in one test.
1. /api/auth
2. ensure user exists in db
3. /api/submit
4. ensure there's a submission in db
5. /api/leaderboard
6. check leaderboard